src/winetricks

```diff
@@ -1078,14 +1078,58 @@ winetricks_get_file_arch()
             *)        w_die "Unknown file arch: ${_W_lipo_output}" ;;
         esac
     else
-        # Assume ELF binaries for everything else
+        # Assume ELF binaries for everything else (or try workaround for some specific scripts)
         _W_ob_output="$(od -An -t x1 -j 0x12 -N 1 "${_W_file}" | tr -d "[:space:]")"
         case "${_W_ob_output}" in
             "3e")      _W_file_arch="x86_64" ;;
             "03"|"06") _W_file_arch="i386" ;;
             "b7")      _W_file_arch="aarch64" ;;
             "28")      _W_file_arch="aarch32" ;;
-            *)         w_die "Unknown file arch: ${_W_ob_output}";;
+            *)  # wineserver and wine might be wrapper scripts, their binaries might be in Wine's bindir.
+                # Since wineboot often is a link next to wineserver (whose path we already know) pointing to
+                # wineapploader in Wine's bindir, we can use this to figure out Wine's bindir.
+                WINEBOOT_BIN="$(echo "${WINESERVER_BIN}"|sed 's,wineserver,wineboot,')"
```

Comment:
It's unlikely to happen in practice, but that wouldn't work if the binary is in a (sub)folder that has the work wineserver in it (e.g., /opt/wineserver-testing).

You could use WINEBOOT_BIN="$(dirname ${WINESERVER_BIN})/wineboot" instead.